### ISSUE TITLE
Mod and Play (1127460)

### ISSUE THREAD

**Issue by Meister1593** [ValveSoftware/Proton#3788]
# Compatibility Report
- Name of the game with compatibility issues: Mod & Play 
**(Unreal engine 4 game)**

- Steam AppID of the game: 1127460

## System Information
- Distribution: Arch rolling
- GPU: Gigabyte GTX 1060 (3 GB)
- Driver/LLVM version: nvidia vulkan 440.66.09
- Kernel version: 5.6.4-arch1-1
- Link to full system information report as [Gist](https://gist.github.com/Meister1593/ca6f8d4c0ebf69dbe00e83b916f7ec08)
- Proton version: Basically any, 4.11-13, 5.0-6, 5.4-GE-3, 5.5-GE-1, 5.6-GE-2

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1127460.log](https://github.com/ValveSoftware/Proton/files/4511781/steam-1127460.log)

## Symptoms 

Game does launch okay, but loading basically any level, including in multiplayer causes game to drop back to menu. It also shows level for a fraction of a second, and instantly drops to menu.
Disabling esync, fsync as well as changing driver to usual (not vulkan branch) does not help. 
Enabling `PROTON_USE_WINED3D` does not help too.
**Interesting thing that game did run before last engine update (i think it's important)**

## Reproduction

1. Launch game
2. Try to start or join game

**Comment by kisak-valve** [comment]
Possible line of interest from the log: `err:steamclient:create_win_interface Don't recognize interface name: SteamNetworkingSockets004`

**Comment by Meister1593** [comment]
> Possible line of interest from the log: `err:steamclient:create_win_interface Don't recognize interface name: SteamNetworkingSockets004`

Yeah i noticed that too, but i don't see how this could be resolved.
This could be resolved manually somehow?

**Comment by Yogfan800** [comment]
I have found 2 ways to get single player to work. Only downside is that workshop items don't work.

1) Run game with Proton 4.2-9
2) Run ModandPlay-Win64-Shipping.exe located in the games installation folder at /modandplay/modandplay/binaries/Win64.

It definitely seems like the SteamNetworkingSockets004 thing you guys mention could be causing the issue.

**Comment by stevenlafl** [comment]
> Possible line of interest from the log: `err:steamclient:create_win_interface Don't recognize interface name: SteamNetworkingSockets004`

Can someone with the game try this as my attempt at getting the SteamNetworkingSockets004 interface implemented: https://github.com/stevenlafl/Proton/releases/tag/proton-5.0-7-sockets004 ?
I don't have the game, but this might do the trick. If not, please supply a proton log with any followup.

If you need it recompiled for 4.2.9 feel free to fork for others or let me know.

Edit: Backlink to PR https://github.com/ValveSoftware/Proton/pull/3887

**Comment by Meister1593** [comment]
> Can someone with the game try this as my attempt at getting the SteamNetworkingSockets004 interface implemented: https://github.com/stevenlafl/Proton/releases/tag/proton-5.0-7-sockets004 ?
> I don't have the game, but this might do the trick. If not, please supply a proton log with any followup.
> 
> If you need it recompiled for 4.2.9 feel free to fork for others or let me know.

It seems that game now works with this version. 
Downloading addons from workshop, joining multiplayer game, creating local game, all works.

**Comment by stevenlafl** [comment]
Thank you!
